app.js

```diff
@@ -1,15 +1,24 @@
 const express = require("express");
-const { getCategoriesC } = require("./controllers/controllers");
+const { getCategoriesC, getReviewsByIdC } = require("./controllers/controllers");
 const error_404_msg = "Sorry can't find that!";
+const error_400_msg = "invalid type please check your input"
+
 const app = express();
 app.use(express.json());
 
 app.get("/api/categories", getCategoriesC);
+app.get("/api/reviews/:review_id", getReviewsByIdC);
 
 app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send(error_400_msg);
+  }
+  if (err.status === 404) {
+    res.status(404).send(error_404_msg);
+  }
   res
     .status(500)
-    .send({ message: "trash server, trash code so you get an Erorr" });
+    .send({ message: "trash server, trash code so you get an Error" });
 });
 
 app.use((req, res, next) => {
```

Comment:
I would place this under the routes as it isn't strictly an error handling middleware function. You could also specify that you would want this error to run for any request to any other endpoint. This can be specified in the first argument of the controller function.